### PR TITLE
Factor out duplicated `entry_point_type` functions

### PR DESCRIPTION
**Pull request by nnethercote**
A small but nice cleanup.

### CODE REVIEW

File: compiler/rustc_passes/src/entry.rs [hunk 1]
```diff
@@ -85,7 +60,11 @@ fn attr_span_by_symbol(ctxt: &EntryContext<'_>, id: ItemId, sym: Symbol) -> Opti
 fn find_item(id: ItemId, ctxt: &mut EntryContext<'_>) {
     let at_root = ctxt.tcx.opt_local_parent(id.owner_id.def_id) == Some(CRATE_DEF_ID);
 
-    match entry_point_type(ctxt, id, at_root) {
+    let attrs = ctxt.tcx.hir().attrs(id.hir_id());
+    let entry_point_type = rustc_ast::entry::entry_point_type(attrs, at_root, || {
+        ctxt.tcx.opt_item_name(id.owner_id.to_def_id())
```
Comment (cjgillot) [inline compiler/rustc_passes/src/entry.rs:65]: `get_name` does not really need to be a closure. Passing an `Option<Symbol>` directly is sufficient. `opt_item_name` only looks at def-path information, and does not involve any query. It should be cheap enough to call unconditionally.

Comment (nnethercote) [reply compiler/rustc_passes/src/entry.rs:65]: Ok.

**Comment by nnethercote** [comment]
I updated the code to remove the closure.

**Comment by cjgillot** [comment]
Thanks.
@bors r+

**Pull request merged**